dpc/src/parameters/testnet1.rs

```diff
@@ -127,12 +126,12 @@ impl Parameters for Testnet1Parameters {
     type AccountSignatureGadget = SchnorrGadget<EdwardsBls12, Self::InnerScalarField, EdwardsBls12Gadget>;
     type AccountSignaturePublicKey = <Self::AccountSignatureScheme as SignatureScheme>::PublicKey;
 
-    type EncryptedRecordCRH = PoseidonCryptoHash<Self::InnerScalarField, 4, false>;
-    type EncryptedRecordCRHGadget = PoseidonCryptoHashGadget<Self::InnerScalarField, 4, false>;
+    type EncryptedRecordCRH = BHPCompressedCRH<EdwardsBls12, 72, 32>;
+    type EncryptedRecordCRHGadget = BHPCompressedCRHGadget<EdwardsBls12, Self::InnerScalarField, EdwardsBls12Gadget, 72, 32>;
     type EncryptedRecordDigest = <Self::EncryptedRecordCRH as CRH>::Output;
 
-    type InnerCircuitIDCRH = PoseidonCryptoHash<Self::OuterScalarField, 4, false>;
-    type InnerCircuitIDCRHGadget = PoseidonCryptoHashGadget<Self::OuterScalarField, 4, false>;
+    type InnerCircuitIDCRH = BHPCompressedCRH<EdwardsBW6, 296, 32>;
+    type InnerCircuitIDCRHGadget = BHPCompressedCRHGadget<EdwardsBW6, Self::OuterScalarField, EdwardsBW6Gadget, 296, 32>;
```

Comment:
The InnerCircuit actually has a pretty long verifying key, because the input is long. We currently has no idea how to handle this.

Reply:
If we compute (part of) the public input MSM outside, then we don't need to include those group elements in the hash